model_compression_toolkit/common/gptq/gptq_training.py

```diff
@@ -0,0 +1,100 @@
+        Use the dataset generator to pass images through the teacher and student networks to get intermediate
+        layers outputs. Use the outputs to compute the observed loss and to back-propagate the error
+        in the student network, to minimize it in the next similar steps.
+        All parameters (such as number of iterations, optimizer, etc.) are in GradientPTQConfig.
+        Args:
+            graph_float: Graph to build a float networks from.
+            graph_quant: Graph to build a quantized networks from.
+            gptq_config: GradientPTQConfig with parameters about the tuning process.
+            fw_info: Framework information
+        """
+        self.fw_info = fw_info
+        self.fw_impl = fw_impl
+        self.graph_float = graph_float
+        self.graph_quant = graph_quant
+        self.gptq_config = gptq_config
+
+        # ----------------------------------------------
+        # Build two models and create compare nodes
+        # ----------------------------------------------
+        self.compare_points, _, self.compare_points_mean, self.compare_points_std = get_compare_points(self.graph_float)
+
+        self.float_model, self.float_user_info = fw_impl.model_builder(self.graph_float,
+                                                          mode=ModelBuilderMode.FLOAT,
+                                                          append2output=self.compare_points,
+                                                          fw_info=fw_info)
+
+        self.fxp_model, self.gptq_user_info = fw_impl.model_builder(self.graph_quant,
+                                                       mode=ModelBuilderMode.GPTQ,
+                                                       append2output=self.compare_points,
+                                                       fw_info=fw_info,
+                                                       gptq_config=gptq_config)
+
+    @abstractmethod
+    def train(self, representative_data_gen: Callable):
+        """
+        Train the quantized model using GPTQ training process
+        Args:
+            representative_data_gen: Dataset to use for inputs of the models.
+        """
+        raise NotImplemented(f'{self.__class__.__name__} have to implement the '
+                             f'framework\'s train method.')
+
+    @abstractmethod
+    def update_graph(self) -> Graph:
+        """
+        Update a graph using GPTQ after minimizing the loss between the float model's output
+        and the quantized model's outputs.
+        Returns:
+            Updated graph after GPTQ.
+        """
+        raise NotImplemented(f'{self.__class__.__name__} have to implement the '
+                             f'framework\'s update_graph method.')
+
+
+def gptq_train(gptq_trainer: GPTQTrainer, representative_data_gen: Callable) -> Graph:
+    """
+    GPTQ training process using knowledge distillation: A teacher network (float model) and a student network (quantized model).
```

Comment:
This function should be called from common and not framework implementation  and have the following API
def gptq_training(gptq_trainer,representative_data_gen,graph_float, graph_quant, gptq_config, fw_imp, fw_info):


Framework implementation should return only GPTQTrainer (uninitlized)

Reply:
arg gptq_trainer is not needed we can get it from fw_imp

Reply:
Good point

Reply:
Done